org.eclipse.xtext/src/org/eclipse/xtext/serializer/analysis/SyntacticSequencerPDAProvider.java

```diff
@@ -454,9 +454,33 @@ public AbstractElementAlias getAmbiguousSyntax() {
 				return ambiguousSyntax;
 			Nfa<ISynState> nfa = getAmbiguousNfa();
 			NfaToProduction prod = new NfaToProduction();
-			ambiguousSyntax = prod.nfaToGrammar(nfa, new GetGrammarElement(), new GrammarAliasFactory());
```

Comment:
we have no tests for this?

Reply:
Not that I'm aware of. The instable generation manifested in a more involved grammar context.